src/btree/bt_sync_obsolete.c

```diff
@@ -391,6 +407,24 @@ __checkpoint_cleanup_page_skip(
         return (0);
     }
 
+    if (__sync_obsolete_limit_reached(session)) {
+        *skipp = true;
+        return (0);
+    }
+
+    /*
+     * Read the page on disk if all the content of the page is not fully deleted and the most recent
+     * transaction is globally visible.
+     */
+    newest_ts = WT_MAX(addr.ta.newest_start_durable_ts, addr.ta.newest_stop_durable_ts);
+    if (!WT_TIME_AGGREGATE_HAS_STOP(&addr.ta) &&
+      __wt_txn_newest_visible_all(session, addr.ta.newest_txn, newest_ts)) {
+        __wt_verbose_debug2(session, WT_VERB_CHECKPOINT_CLEANUP,
+          "%p: obsolete time window page read into the cache", (void *)ref);
+        WT_STAT_CONN_INCR(session, checkpoint_cleanup_obsolete_tw_pages_read);
```

Comment:
We only increase the stat but do nothing else?

Reply:
I understand what it is. Should we increase the stat at the caller instead of here?

Reply:
The skipped stats can also be updated in the caller. We only need to do that in one place?